back/db/views/analytics_fact_participations_v01.sql

```diff
@@ -0,0 +1,63 @@
+-- Analytics view for all types of participation
+SELECT
+    i.id,
+    i.author_id AS user_id,
+    i.project_id,
+    adt.id as type_id,
+    i.created_at::DATE AS created_date,
+    upvotes_count + downvotes_count as votes_count,
+    upvotes_count,
+    downvotes_count
+from ideas i
+INNER JOIN analytics_dimension_types adt ON adt.name = 'idea'
+
+UNION ALL
+
+SELECT
+    i.id,
+    i.author_id AS user_id,
+    null AS project_id, -- initiative has no project
+    adt.id as type_id,
+    i.created_at::DATE AS created_date,
+    upvotes_count + downvotes_count as votes_count,
+    upvotes_count,
+    downvotes_count
+FROM initiatives i
+INNER JOIN analytics_dimension_types adt ON adt.name = 'initiative'
+
+UNION ALL
+
+SELECT
+    c.id,
+    c.author_id AS user_id,
+    i.project_id as project_id,
+    adt.id as type_id,
+    c.created_at::DATE AS created_date,
+    c.upvotes_count + c.downvotes_count as votes_count,
+    c.upvotes_count,
+    c.downvotes_count
+FROM comments c
+INNER JOIN analytics_dimension_types adt ON adt.name = 'comment'
+INNER JOIN analytics_dimension_dates add ON add.date = c.created_at::DATE
```

Comment:
I don't understand that inner join. What is it for?

Reply:
each fact table should contain ids to join them to their dimension tables where we can get further data, like the parent of a type, or the year of a date. its not supposed to be well normalized
@jamesspeake in the case of the date we could avoid doing the joing given that the primary key its actually the date right?

Reply:
What I don't understand is why we check that the creation date of the comment exists in the date dimension table, but we don't do it for initiatives and ideas.

Reply:
oh now i see what you mean, don't know really since we are not using participations for now so i haven't started working with this